### PR TITLE
[demos] Real-time feature engineering demo.

### PR DESCRIPTION
**Pull request by ryzhyk**
The demo is similar to the fraud detection demo, but with a slightly different focus:

* Demonstrates several different SQL constructs and how they can be composed by stacking simple views.

* Demos bounded-state time series analysis.

* Demos strong consistency.

### CODE REVIEW

**Review by lalithsuresh — APPROVED** [review]

**Comment by ryzhyk** [comment]
This demo will run for a relatively long time and consume lots of CPU resources (roughly 100% of 8 cores).

@lalithsuresh , would you like to rate-limit it?

**Comment by lalithsuresh** [comment]
@ryzhyk yes, let's do the usual throttling to keep arrivals rates down to 1000 events/sec.

File: demo/packaged/sql/03-feature-engineering.sql [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+-- * Data enrichment using ASOF JOIN
+-- * Rolling aggregate over a 30-day time window
+-- * Global aggregates over the entire event history
+--
+-- As you browse through the code, note how complex features are implemented by composing simple
+-- SQL views.
+--
+-- Try it!
+-- * ▶️ Start the demo to generate and process a stream of 1 billion random credit card transactions.
+-- * 📈 Check the Performance tab: as Feldera processes millions of incoming transactions,
```
Comment (lalithsuresh) [inline demo/packaged/sql/03-feature-engineering.sql:15]: One more suggestion. Can we use multi-line comments (/* */)? It looks less distracting in our UI.

File: demo/packaged/sql/03-feature-engineering.sql [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+▶️ Start the demo to generate and process a stream of 1 billion random credit card transactions.
+📈 Check the Performance tab: as Feldera processes millions of incoming transactions,
+   its memory footprint remains steady around the 2GB mark.
+   See https://docs.feldera.com/tutorials/time-series to learn how Feldera evaluates many complex
+   SQL queries using bounded memory.
+💪 Feldera is strongly consistent: its output correctly reflects all inputs received so far.
+   Scroll down to the 'user_stats' view to learn more.
+*/
+
+-- Generate 10000 random customer records.
```
Comment (mihaibudiu) [inline demo/packaged/sql/03-feature-engineering.sql:25]: this comment should be in the connectors section

File: demo/packaged/sql/03-feature-engineering.sql [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+          {
+            "limit": 1,
+            "fields": {"ts": { "values": ["2080-01-01T00:00:00Z"] } }
+          }]
+        }
+      }
+    }]'
+);
+
+-- Generate a stream of 1 billon credit card transactions.
```
Comment (mihaibudiu) [inline demo/packaged/sql/03-feature-engineering.sql:54]: typo in billon

File: demo/packaged/sql/03-feature-engineering.sql [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+-- Rolling aggregation: Compute the number of out-of-state transactions in the last 30 days for each transaction.
+CREATE VIEW transaction_with_history AS
+SELECT
+    *,
+    COUNT(case when out_of_state then 1 else 0 end) OVER window_30_day as out_of_state_count
+FROM
+    enriched_transaction
+WINDOW window_30_day AS (PARTITION BY customer_id ORDER BY ts RANGE BETWEEN INTERVAL 30 DAYS PRECEDING AND CURRENT ROW);
+
+-- Flag out-of-state transactions, but only if there have been fewer than 5 such transactions
```
Comment (mihaibudiu) [inline demo/packaged/sql/03-feature-engineering.sql:104]: My talk has a video describing a demo very similar

**Pull request merged**